python/OnnxUdf/onnx_example.ipynb

```diff
@@ -0,0 +1,192 @@
+    "    bands=[\"B04\"],\n",
+    "    max_cloud_cover=20,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After setting up the connection with a back-end, we define where both the dependencies and the model are stored.\n",
+    "\n",
+    "### Dependencies\n",
+    "In order to remain lightweight, an openEO backend does not have the necessary dependencies to run the model. Therefore, we need to pass the extra dependencies to the backend. If you are using an ONNX model that can be run with ONNX Runtime version 1.16.3, you can use the dependencies url provided in the code below. If you are using a different version of ONNX Runtime, you need to provide the dependencies yourself.\n",
+    "\n",
+    "More information on onnx version compatibility can be found here: https://github.com/onnx/tutorials#converting-to-onnx-format\n",
+    "\n",
+    "More information on how to make a dependency archive for openeo can be found here: https://documentation.dataspace.copernicus.eu/APIs/openEO/job_config.html#custom-udf-dependencies\n",
+    "\n",
+    "### Model\n",
+    "The model needs to be stored in a location that is accessible by the backend. In this example, we store the model in the same location as the dependencies. You could however also store the model in a public github repository and provide the url to the backend."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# This will be required to import onnxruntime in the UDF.\n",
+    "dependencies_url = \"https://artifactory.vgt.vito.be:443/auxdata-public/openeo/onnx_dependencies_1.16.3.zip\"\n",
+    "# You can upload your own model to a public download link and paste the url here. An example of a public storage is github, or you can set up a public S3 bucket.\n",
+    "model_url = \"https://artifactory.vgt.vito.be:443/auxdata-public/openeo/test_onnx_model.zip\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next we load the udf that will run the model. Make sure to take a look at [the udf code](./onnx_udf.py) to understand how the loading of the model is cached, how you can work with the named dimensions of an Xarray and how to run a model on each separate timestep.\n",
+    "\n",
+    "We use the process `apply_neighborhood` to apply the udf. The model accepts a window of 256x256 pixels. In order to get a smooth result, we apply the model on overlapping windows. The values included in the overlap will not be modified.\n",
+    "In this case an overlap of 16 pixels is used on both sides, so the remaining size of 224x224 pixels will be modified each step.\n",
+    "\n",
+    "Other models might perform pixel-wise predictions, in which case you can use the `reduce_dimension` process to only keep the predictions. The openEO backend will then automatically handle the chunking and stitching of the results."
```

Comment:
Here we should be cautious, you can also have networks who work on single bands/timesteps. So there is not necessarily a reduced dimension

Reply:
Perhaps I could list some more alternatives. It depends if the bands need to be kept after the prediction or not.

Reply:
Maybe, although I do not think this is the best place to document all options. Maybe its better to give a specific example:

e.g. In case an other model performs a pixel wise prediction causing a collapse of one of the dimensions of the input, ....

or something along this trend.